lib/unleash/feature_toggle.rb

```diff
@@ -51,33 +54,45 @@ def self.disabled_variant
 
     private
 
-    def resolve_variant(context, toggle_enabled)
-      return Unleash::FeatureToggle.disabled_variant unless toggle_enabled
-      return Unleash::FeatureToggle.disabled_variant if sum_variant_defs_weights <= 0
+    def resolve_variant(context, evaluation_result, group_id)
+      variant_definitions = evaluation_result[:strategy]&.variant_definitions || self.variant_definitions
+      variant_definitions = self.variant_definitions if variant_definitions.empty?
```

Comment:
nit pick/bike shedding for legibility:

(only one override of values, with all criterias in one line)
```suggestion
      variant_definitions = evaluation_result[:strategy]&.variant_definitions
      variant_definitions = self.variant_definitions if variant_definitions.nil? || variant_definitions.empty?
```